### PR TITLE
RAC-4534 - Validate the general configuration of the installed components

### PR DESCRIPTION
**Pull request by larry-dean**
* Pacemakerr status and configuration
* Mongo status and configuration
* Rabbit status and configuration
* RackHD components

### CODE REVIEW

**Comment by larry-dean** [comment]
@patelb10 @johren @hohene

File: test/tests/ha/rackdh_ha_resource_test.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+# LOCAL
+from nose.plugins.attrib import attr
+from nosedep import depends
+
+FAIL_ON_NODE_COLLECTION = True
+active_node_list = list()
+
+
+@attr(regression=False, smoke=False, ha_resource_tests=True)
+class HaResourceTest(fit_common.unittest.TestCase):
```
Comment (hohene) [inline test/tests/ha/rackdh_ha_resource_test.py]: take fit_common. off.  ie. (unittest.TestCase).  Thought we are trying to remove that level of referencing.

File: test/tests/ha/rackdh_ha_resource_test.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
+"""
+Copyright 2017, Dell, Inc.
+
+This test checks the installation of the HA environment
+"""
+
+import fit_common
```
Comment (hohene) [inline test/tests/ha/rackdh_ha_resource_test.py]: import fit_path  # NOQA: unused import
import unittest

File: test/tests/ha/rackdh_ha_resource_test.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
+"""
+Copyright 2017, Dell, Inc.
```
Comment (hohene) [inline test/tests/ha/rackdh_ha_resource_test.py]: Update the copyright, "Copyright (c) 2017 Dell Inc. or its subsidiaries.  All Rights Reserved."
I need to fix the template!

File: test/tests/ha/rackdh_ha_resource_test.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        for vmnum in range(1, self.number_of_vms + 1):
+            # Check for default configuration
+            self.assertFalse(self.has_online_nodes(vmnum, 'ora'),
+                             "Default configuration on vm {}".format(vmnum))
+
+    def test012_check_node_status(self):
+        for vmnum in range(1, self.number_of_vms + 1):
+            for node_number in range(1, self.number_of_vms + 1):
+                # Determine if this node is online
+                self.assertTrue(self.has_online_nodes(vmnum, 'node{}'.format(node_number)),
```
Comment (hohene) [inline test/tests/ha/rackdh_ha_resource_test.py]: Do you want to exit the test on the first failure here, or do you want to check each vmnum/node combo an report back a list of failures?
if so, I suggest creating an errorlist = [], append the error to the errorlist.
When done, check if errorlist is not empty, dump the list of nodes/errors that are not online and then assert.
Then you can see what is failing.

File: test/tests/ha/rackdh_ha_resource_test.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+                                "Node {} is not online on vm {}".format(node_number, vmnum))
+
+    def test020_check_and_collect_online_nodes(self):
+        # determine node status and collect active nodes in cluster
+        for vmnum in range(1, self.number_of_vms + 1):
+            command = "crm_mon -X | grep 'node{}.*online=.true' -q".format(vmnum)
+            status = fit_common.remote_shell(command, vmnum=vmnum)['exitcode']
+            if FAIL_ON_NODE_COLLECTION:
+                self.assertEqual(status, 0, msg='node{} is offline'.format(vmnum))
```
Comment (hohene) [inline test/tests/ha/rackdh_ha_resource_test.py]: Same comment as above and in subsequent tests.

**Comment by larry-dean** [comment]
Changes to address @hohene comments pushed up.

File: test/tests/ha/rackdh_ha_resource_test.py [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    def test031_check_mongo_colocation(self):
+        err = ''
+        for mongo_number in range(1, self.number_of_services + 1):
+            command = "crm config show | grep 'mongo{0}.*inf:.*mongo_{0}.*addr_{0}'" \
+                      .format(mongo_number)
+            status = fit_common.remote_shell(command, vmnum=active_node_list[0])['exitcode']
+            if status != 0:
+                err += 'docker_mongo_{0} on mongo_addr_{0}, '.format(mongo_number)
+        self.assertEqual(len(err), 0, 'The following Mongo coLocations are missing: {}'
+                         .format(err.rstrip(", ")))
```
Comment (patelb10) [inline test/tests/ha/rackdh_ha_resource_test.py]: I think we also have to check resource "mongo_addr_" before checking colocation between 'docker_mongo_' and 'mongo_addr_'

**Comment by PengTian0** [comment]
test this please

**Comment by larry-dean** [comment]
Added tests to address @patelb10 concerns.

**Comment by larry-dean** [comment]
Added validation of mongo and rabbit IP addresses

**Comment by johren** [comment]
test this please

**Comment by larry-dean** [comment]
test this please

**Comment by larry-dean** [comment]
Push change to rabbit anti-colocation error message and changed line separator from windows (cr/lf) to linux (lf)

**Pull request merged**